### PR TITLE
Port type moves to UWP 6.2

### PR DESCRIPTION
**Pull request by morganbr**
The .NET Native CoreLib already has the types that moved in master (Hashtable and ConcurrentQueue), so we need CoreFX to include the matching changes. Also includes porting a couple other changes that the ports depended on.

Depends on https://github.com/dotnet/coreclr/pull/18818 so the uap flavor builds too.

### CODE REVIEW

File: src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj [hunk 1]
```diff
@@ -76,6 +76,18 @@
     <Compile Include="$(CommonPath)\System\HResults.cs">
       <Link>Common\System\HResults.cs</Link>
     </Compile>
+    <Compile Include="$(CommonPath)\CoreLib\System\Text\ValueStringBuilder.cs">
+      <Link>CoreLib\System\Text\ValueStringBuilder.cs</Link>
+    </Compile>
+  </ItemGroup>
+  <!-- TODO: Remove these lines when a new uapaot build is ingested into corefx. -->
+  <ItemGroup Condition="'$(TargetGroup)' == 'uapaot'">
```
[the remark this answers is not in the document]
Comment (morganbr) [reply src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj]: I think you reviewed one of the intermediate commits. That shouldn't be in the final version.

[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/System.Runtime.Extensions/src/System.Runtime.Extensions.csproj]: Right. You added Jans commit that deletes these lines afterwards. Overlooked it. Thanks.

**Comment by zamont** [comment]
@morganbr my PR #31663 updates to the latest .NET Native CoreLib and the CI build fails with the Hashtable existing in both places. How do you want to move forward?

**Comment by zamont** [comment]
I appended my dependency updates from #31663 to the tip of this PR, and the resulting CI should be green.

**Comment by zamont** [comment]
@dotnet-bot test NETFX x86 Release Build

**Comment by zamont** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by zamont** [comment]
@MattGal the [PR test run](https://mc.dot.net/#/user/morganbr/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Frelease~2Fuwp6.2~2F/test~2Ffunctional~2Fcli~2F/ed66481e14da5d01c95852d598b6de3aec058d2a) shows Fedora.26.Amd64.Open-x64 on fire. Is this new?

**Comment by MattGal** [comment]
@zamont just sent mail again, please see me if you need help.  I notified folks that this OS was end-of-life and that this would begin happening yesterday last Thursday.

**Comment by zamont** [comment]
@morganbr this change looks ready to merge and we'll need this change to make progress, since the types only line up with both the updated .NET Native dependency and your moving of types

cc: @tommcdon

**Comment by zamont** [comment]
@MichalStrehovsky when can we move forward with this set of changes? I believe Morgan is also out this week

**Comment by MichalStrehovsky** [comment]
> @MichalStrehovsky when can we move forward with this set of changes?

Since these changes are required to unblock the ingestion at the .NET Native side, I would say ASAP? What help do you need from me on this?

**Comment by zamont** [comment]
Ok. I will go ahead and merge this then.

**Comment by morganbr** [comment]
@zamont, the reason I hadn't merged this is that we didn't have a build of CoreCLR that matched it. If you have that now, merging was fine. Thanks for taking care of it.

**Comment by zamont** [comment]
Yup, we've got our uwp6.2 builds up and running so we were able to produce a new CoreCLR build with matching CoreLib. Awesome :)

**Pull request merged**